Remotion/Data/DomainObjects.UnitTests/TestDomainDatabase.cs

```diff
@@ -0,0 +1,104 @@
+          TableInheritanceConfiguration.Initialize();
+
+          SqlConnection.ClearAllPools();
+
+          var masterAgent = new DatabaseAgent(DatabaseTest.MasterConnectionString);
+          masterAgent.ExecuteBatchFile("Database\\DataDomainObjects_CreateDB.sql", false, DatabaseConfiguration.GetReplacementDictionary());
+          var testDomainAgent = new DatabaseAgent(DatabaseTest.TestDomainConnectionString);
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TestDomain_TearDownDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TableInheritanceTestDomain_TearDownDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TestDomain_SetupDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TableInheritanceTestDomain_SetupDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_Patches.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+
+          s_standardMappingDatabaseAgent = new StandardMappingDatabaseAgent(DatabaseTest.TestDomainConnectionString);
+          string sqlFileName = StandardMappingTest.CreateTestDataFileName;
+          s_standardMappingDatabaseAgent.ExecuteBatchFile(sqlFileName, true, DatabaseConfiguration.GetReplacementDictionary());
+          string sqlFileName1 = TableInheritanceMappingTest.CreateTestDataFileName;
+          s_standardMappingDatabaseAgent.ExecuteBatchFile(sqlFileName1, true, DatabaseConfiguration.GetReplacementDictionary());
+
+          // We don't want the tests to initialize a default mapping; therefore, modify MappingConfiguration.s_fields.Current so that it will 
+          // throw when asked to generate a new MappingConfiguration.
+
+          var throwingMappingConfigurationContainer = new DoubleCheckedLockingContainer<IMappingConfiguration>(
+              () =>
+              {
+                throw new InvalidOperationException(
+                    "This test failed to setup the mapping configuration. Did you forget to derive from StandardMappingTest or to call base.SetUp?");
+              });
+          var fields = PrivateInvoke.GetNonPublicStaticField(typeof(MappingConfiguration), "s_fields");
+          Assertion.IsNotNull(fields);
+          PrivateInvoke.SetPublicField(fields, "Current", throwingMappingConfigurationContainer);
+        }
+        catch (Exception ex)
+        {
+          Console.WriteLine("SetUpFixture failed: " + ex);
+          Console.WriteLine();
+          throw;
+        }
+      }
+    }
+
+    public static void CleanupDatabase ()
+    {
+      if (!s_initialized)
+        return;
+
+      s_standardMappingDatabaseAgent.SetDatabaseReadWrite(DatabaseTest.DatabaseName);
```

Comment:
We no longer call s_standardMappingDatabaseAgent.SetDatabaseReadOnly(). -> we can remove this here, too. Might result in a little more performance.